src/execution/execute-afm.convert.ts

```diff
@@ -7,32 +7,78 @@ function convertElementsArray(arr: string[], isText?: boolean): ExecuteAFM.Attri
     return isText ? { values: arr } : { uris: arr };
 }
 
+function convertPositiveAttributeFilter(filter: AFM.IPositiveAttributeFilter): ExecuteAFM.IPositiveAttributeFilter {
+    const { positiveAttributeFilter: oldFilter } = filter;
+
+    return {
+        positiveAttributeFilter: {
+            displayForm: oldFilter.displayForm,
+            in: convertElementsArray(oldFilter.in, oldFilter.textFilter)
+        }
+    };
+}
+
+function convertNegativeAttributeFilter(filter: AFM.INegativeAttributeFilter): ExecuteAFM.INegativeAttributeFilter {
+    const { negativeAttributeFilter: oldFilter } = filter;
+
+    return {
+        negativeAttributeFilter: {
+            displayForm: oldFilter.displayForm,
+            notIn: convertElementsArray(oldFilter.notIn, oldFilter.textFilter)
+        }
+    };
+}
+
 function convertFilter(filter: AFM.CompatibilityFilter): ExecuteAFM.CompatibilityFilter {
     if (AFM.isPositiveAttributeFilter(filter)) {
-        const { positiveAttributeFilter: oldFilter } = filter;
+        return convertPositiveAttributeFilter(filter);
+    } else if (AFM.isNegativeAttributeFilter(filter)) {
+        return convertNegativeAttributeFilter(filter);
+    }
 
-        return {
-            positiveAttributeFilter: {
-                displayForm: oldFilter.displayForm,
-                in: convertElementsArray(oldFilter.in, oldFilter.textFilter)
-            }
-        };
+    return filter;
+}
+
+function convertMeasureFilter(filter: AFM.FilterItem): ExecuteAFM.FilterItem {
+    if (AFM.isPositiveAttributeFilter(filter)) {
+        return convertPositiveAttributeFilter(filter);
     } else if (AFM.isNegativeAttributeFilter(filter)) {
-        const { negativeAttributeFilter: oldFilter } = filter;
+        return convertNegativeAttributeFilter(filter);
+    }
+
+    return filter;
+}
+
+function convertFilters(filters?: AFM.CompatibilityFilter[]): ExecuteAFM.CompatibilityFilter[] | undefined {
+    return filters !== undefined ? filters.map(convertFilter) : filters;
+}
+
+function convertMeasureFilters(filters?: AFM.FilterItem[]): ExecuteAFM.FilterItem[] | undefined {
+    return filters !== undefined ? filters.map(convertMeasureFilter) : filters;
+}
+
+function convertMeasure(measure: AFM.IMeasure): ExecuteAFM.IMeasure {
+    if (AFM.isSimpleMeasureDefinition(measure.definition)) {
+        const simpleMeasure = measure.definition.measure;
+        const filters = convertMeasureFilters(simpleMeasure.filters);
+        const filtersProp = filters ? { filters } : {};
```

Comment:
please consider `lodash/isEmpty` which also ensure that empty filters are not passed 
```
const filtersProp = !isEmpty(filters) ? { filters } : {};
```

Reply:
regardless of the isEmpty() the empty filters will be passed because they are expanded from simpleMeasure definition

Reply:
to be honest, i would keep it as is - have responsibility of the converter to be just flipping the structures (filters) to ExecuteAFM and avoid sanitization (that should be done earlier on the AFM itself)